### PR TITLE
refactor: Use protobuf struct to perform device authentication

### PR DESCRIPTION
**Pull request by kartikCypherock**
Implements https://app.clickup.com/t/9002019994/PRF-4709

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4709 Device authentication](https://app.clickup.com/t/9002019994/PRF-4709)

File: src/constant_texts.c [hunk 1]
```diff
@@ -115,6 +115,14 @@ const char *ui_text_options_buzzer_adjust[] = {
 };
 #endif /* DEV_BUILD */
 
+// Manager app text
+// Device authentication text
+const char *ui_text_message_device_authenticating = "Device authenticating..";
```
[the remark this answers is not in the document]
Comment (amanCypherock) [reply src/constant_texts.c:120]: Make sure you've taken this from figma, Texts were inconsistent before.

File: src/level_four/core/controller/device_authentication_controller.c [hunk 2]
```diff
@@ -130,259 +110,220 @@ static void helper_get_gendig_hash(atecc_slot_define_t slot,
...
-      atca_sign_internal_in_out_t sign_internal_param = {0};
-      get_io_protection_key(io_protection_key);
+manager_auth_device_response_t __attribute__((optimize("O0")))
+sign_serial_number(void) {
+  manager_auth_device_response_t response =
+      MANAGER_AUTH_DEVICE_RESPONSE_INIT_ZERO;
+  response.which_response = MANAGER_AUTH_DEVICE_RESPONSE_SERIAL_SIGNATURE_TAG;
+
+  uint8_t nonce[32] = {0};
+  // auth_data_t auth_serial_packet = {0};
```
Comment (amanCypherock) [inline src/level_four/core/controller/device_authentication_controller.c:124]: Remove comment

Comment (amanCypherock) [reply src/level_four/core/controller/device_authentication_controller.c:124]: In sign challenge funcition and controller main too.

File: src/level_four/core/controller/device_authentication_controller.c [hunk 3]
```diff
@@ -130,259 +110,220 @@ static void helper_get_gendig_hash(atecc_slot_define_t slot,
...
+    sign_internal_param.message = tempkey_hash;
+    sign_internal_param.digest = final_hash;
+
+    helper_sign_internal_msg(&sign_internal_param,
+                             SIGN_MODE_INTERNAL,
+                             slot_2_auth_key,
+                             slot_5_challenge);
+
+    // overwrite challenge slot to signature generation on same challenge
+    memset(challenge, 0, 32);
```
Comment (amanCypherock) [inline src/level_four/core/controller/device_authentication_controller.c:286]: Use macro for challenge and serial size

**Review by ujjwal-cyph — APPROVED** [review]

File: src/constant_texts.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (kartikCypherock) [reply src/constant_texts.c:120]: Only 1 text was not matching, I have fixed that

File: src/level_four/core/controller/device_authentication_controller.c (hunk 2 above)
Comment (kartikCypherock) [reply src/level_four/core/controller/device_authentication_controller.c:124]: Removed

File: src/level_four/core/controller/device_authentication_controller.c (hunk 3 above)
Comment (kartikCypherock) [reply src/level_four/core/controller/device_authentication_controller.c:286]: Ok, added a macro

**Pull request merged**